### PR TITLE
[setup] log init submodules

### LINKED ISSUE #426: Copy FireSim's logging for initial setup scripts
E.g. https://github.com/firesim/firesim/blob/master/build-setup.sh

This makes it easier to debug user issues during setup

### PR DESCRIPTION
**Pull request by abejgonzalez**
**Related issue**: Fixes #426 

**Type of change**: other enhancement

**Impact**: other

**Release Notes**
Log the setup scripts to help users debug (`init-submodules` and `build-toolchains`). Follows the recommendation given in #426.

### CODE REVIEW

**Comment by jerryz123** [comment]
The fireboom/firesim failures are being addressed in #390

**Comment by a0u** [comment]
See:
https://github.com/ucb-bar/chipyard/blob/7bcedfa7ad654f7bbdc61a417dcc39c28ab676f0/scripts/build-util.sh#L50
https://github.com/ucb-bar/chipyard/blob/7bcedfa7ad654f7bbdc61a417dcc39c28ab676f0/scripts/build-util.sh#L83

**Comment by abejgonzalez** [comment]
I think then we can just log the `init-submodules` step since that doesn't have a log. I didn't notice that the `build-utils` made a log. Thoughts?

**Comment by colinschmidt** [comment]
I think we just log the init submodules, and point users to the other log files in that log or the readme/docs/end of build-tools error message.

File: scripts/init-submodules-no-riscv-tools.sh [hunk 1]
```diff
@@ -1,63 +1,8 @@
...
-
-# Only shallow clone needed for basic SW tests
-git submodule update --init software/firemarshal
-
-# Configure firemarshal to know where our firesim installation is
-if [ ! -f $RDIR/software/firemarshal/marshal-config.yaml ]; then
-  echo "firesim-dir: '../../sims/firesim/'" > $RDIR/software/firemarshal/marshal-config.yaml
-fi
-echo "PATH=\$( realpath \$(dirname "\${BASH_SOURCE[0]}") )/software/firemarshal:\$PATH" >> $RDIR/env.sh
+bash scripts/init-submodules-no-riscv-tools-nolog.sh $@ 2>&1 | tee init-submodules-no-riscv-tools.log
```
Comment (a0u) [inline scripts/init-submodules-no-riscv-tools.sh]:
```suggestion
./scripts/init-submodules-no-riscv-tools-nolog.sh "$@" 2>&1 | tee init-submodules-no-riscv-tools.log
```

Quoting is needed so the individual parameters are passed intact, without interpretation or expansion.

**Pull request merged**